client/packages/common/src/hooks/useElectronClient/index.ts

```diff
@@ -0,0 +1,87 @@
+import { useState, useEffect } from 'react';
+
+const DISCOVERY_TIMEOUT = 5000;
+
+// Should match server/server/src/discovery.rs (FrontEndHost)
+export type FrontEndHost = {
+  protocol: 'http' | 'https';
+  port: number;
+  ip: string;
+  name: string;
+  clientVersion: string;
+  // This one is set by electron.ts
+  isLocal: boolean;
+};
+
+type ServerDiscovered = (event: object, value: FrontEndHost) => void;
+export type ElectronAPI = {
+  serverDiscovered: (eventInfo: ServerDiscovered) => void;
+  startServerDiscovery: () => void;
+  connectToServer: (server: FrontEndHost) => void;
+  connectedServer: () => Promise<FrontEndHost | null>;
+};
+
+declare global {
```

Comment:
this file is the only one that uses window.electronAPI, so though it was ok to add global here